container/docker/handler.go

```diff
@@ -294,32 +294,8 @@ func (self *dockerContainerHandler) GetStats() (*info.ContainerStats, error) {
 }
 
 func (self *dockerContainerHandler) ListContainers(listType container.ListType) ([]info.ContainerReference, error) {
-	if self.name != "/docker" {
```

Comment:
The docker ID is extracted from the container name - https://github.com/google/cadvisor/blob/499737d7ad3f754a63036e9c4d0d5f77f2889fed/container/docker/factory.go#L118. It is that ID that is used in the docker inspect.

Again, this is tested & working properly on Ubuntu 14.04.